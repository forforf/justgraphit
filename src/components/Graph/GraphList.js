import React from 'react';
import GraphListItem from "./GraphListItem";


function GraphList({graphNameList, changeSelectedGraph, deleteGraph}) {

  const graphList = graphNameList.map((graphName, index) => (
    <GraphListItem
      key={`graphList-${index}`}
      handleClick={changeSelectedGraph}
      graphName={graphName}
      // deleteGraph={storageFns.deleteGraph}
      deleteGraph={deleteGraph}
    />
  ));

  return (
    <div className="GraphList">
      <h3>Graph List</h3>
      {graphList}
    </div>
  );
}

export default GraphList;
