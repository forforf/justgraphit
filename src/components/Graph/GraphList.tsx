import React from 'react';
import GraphListItem from './GraphListItem';
import { GraphHandler, GraphName } from '../../JustGraphitTypes';

export type GraphListProps = {
  graphNameList: GraphName[];
  changeSelectedGraph: GraphHandler;
  deleteGraph: GraphHandler;
};

function GraphList({
  graphNameList,
  changeSelectedGraph,
  deleteGraph,
}: GraphListProps): JSX.Element {
  const graphNames = graphNameList ?? [];
  const graphList = graphNames.map((graphName, index) => (
    <GraphListItem
      key={`graphList-${index}`}
      handleClick={changeSelectedGraph}
      graphName={graphName}
      // deleteGraph={storageFns.deleteGraph}
      deleteGraph={deleteGraph}
    />
  ));

  return (
    <div className="GraphList">
      <h3>Graph List</h3>
      {graphList}
    </div>
  );
}

export default GraphList;
