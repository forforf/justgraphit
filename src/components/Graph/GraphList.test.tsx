import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import GraphList, {GraphListProps} from './GraphList';


describe('single empty graph', () => {
  let graphListProps: GraphListProps;
  beforeEach(() => {
    graphListProps = {
      graphNameList: ['graph 1'],
      changeSelectedGraph: jest.fn(),
      deleteGraph: jest.fn()
    }
  });

  test('deep snapshot', () => {
    const component = renderer.create(
      <GraphList {...graphListProps}/>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('shallow snapshot', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render( <GraphList {...graphListProps}/>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot()
  });
})