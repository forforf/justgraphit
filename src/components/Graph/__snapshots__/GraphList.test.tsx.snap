// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single empty graph deep snapshot 1`] = `
<div
  className="GraphList"
>
  <h3>
    Graph List
  </h3>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
    onClick={[Function]}
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        graph 1
      </span>
    </div>
    <div
      className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </div>
  </li>
</div>
`;

exports[`single empty graph shallow snapshot 1`] = `
<div
  className="GraphList"
>
  <h3>
    Graph List
  </h3>
  <GraphListItem
    deleteGraph={[MockFunction]}
    graphName="graph 1"
    handleClick={[MockFunction]}
  />
</div>
`;
